src/System.Management.Automation/engine/remoting/fanin/PriorityCollection.cs

```diff
@@ -154,15 +162,29 @@ internal void Add<T>(RemoteDataObject<T> data)
         /// </summary>
         internal void Clear()
         {
-            Dbg.Assert(null != _dataToBeSent, "Serialized streams are not initialized");
-            lock (_syncObjects[(int)DataPriorityType.PromptResponse])
+            /* Lock _dataSyncObjects and _dataToBeSent as a set to ensure atomic cleanup. */
+            lock (_syncObject)
```

Comment:
I think we can avoid the sync object `_syncObject` by the following code:
```CSharp
if (null != _dataSyncObjects)
{
    int promptResponseIndex = (int)DataPriorityType.PromptResponse;
    int defaultIndex = (int)DataPriorityType.Default;

    if (_dataToBeSent[promptResponseIndex] != null)
    {
        lock (_dataSyncObjects[promptResponseIndex])
        {
            if (_dataToBeSent[promptResponseIndex] != null)
            {
                _dataToBeSent[promptResponseIndex].Dispose();
                _dataToBeSent[promptResponseIndex] = null;
            }
        }
    }

    if (_dataToBeSent[defaultIndex] != null)
    {
        lock (_dataSyncObjects[defaultIndex])
        {
            if (_dataToBeSent[defaultIndex] != null)
            {
                _dataToBeSent[defaultIndex].Dispose();
                _dataToBeSent[defaultIndex] = null;
            }
        }
    }
}
```

Reply:
You're right; I'll make the change.

Reply:
fixed